File: app/src/main/java/com/example/myapplication/activities/mangapage/manga_detail_mvp/MangaDetailInterface.java
package com.example.myapplication.activities.mangapage.manga_detail_mvp;

import com.example.myapplication.models.mangamodels.DetailMangaModel;

import java.util.List;

public interface MangaDetailInterface {
    void onGetDetailDataSuccess(DetailMangaModel detailMangaModel);

    void onGetDetailDataFailed();

    void onGetAllChapterSuccess(List<DetailMangaModel.DetailAllChapterDatas> detailAllChapterDatasList);

    void onGetAllChapterFailed();

    void onGetGenreSuccess(List<DetailMangaModel.DetailMangaGenres> genresList);

    void onGetGenreFailed();
}

File: app/src/main/java/com/example/myapplication/activities/mangapage/manga_detail_mvp/MangaDetailInterface.kt
package com.example.myapplication.activities.mangapage.manga_detail_mvp

import com.example.myapplication.models.mangamodels.DetailMangaModel

interface MangaDetailInterface {
    fun onGetDetailDataSuccess(detailMangaModel: DetailMangaModel)

    fun onGetDetailDataFailed()

    fun onGetAllChapterSuccess(detailAllChapterDatasList: List<DetailMangaModel.DetailAllChapterDatas>)

    fun onGetAllChapterFailed()

    fun onGetGenreSuccess(genresList: List<DetailMangaModel.DetailMangaGenres>)

    fun onGetGenreFailed()
}
